src/main/scala/com/fulcrumgenomics/phasing/HapCutToVcf.scala

```diff
@@ -0,0 +1,413 @@
+  @arg(flag="u", doc="The output VCF with variants that were not phased by HAPCUT.") val unphasedOutput: PathToVcf,
+  @arg(flag="m", doc="Allow missing fields in the VCF header.") val allowMissingFieldsInVcfHeader: Boolean = true,
+  @arg(flag="r", doc="The first variant context in a block should have '/' instead of '|'") val readBackedPhasingFormat: Boolean = true
+) extends FgBioTool with LazyLogging {
+  Io.assertReadable(vcf)
+  Io.assertReadable(input)
+  Io.assertCanWriteFile(output)
+
+  override def execute(): Unit = {
+    val vcfReader      = new VCFFileReader(vcf.toFile, false)
+    val phasedWriter   = makeWriter(output, vcfReader, addPhasedHeaderLines=true)
+    val unphasedWriter = makeWriter(unphasedOutput, vcfReader, addPhasedHeaderLines=false)
+
+    val iterator = new MergingIterator(input, vcfReader, readBackedPhasingFormat)
+    iterator.foreach {
+      case Left(ctx)  => phasedWriter.add(ctx)
+      case Right(ctx) => unphasedWriter.add(ctx)
+    }
+
+    CloserUtil.close(vcfReader)
+    phasedWriter.close()
+    unphasedWriter.close()
+  }
+
+  /** Creates a VCF writer, adding extra header lines if the output is the phased VCF. */
+  private def makeWriter(path: PathToVcf, vcfReader: VCFFileReader, addPhasedHeaderLines: Boolean): VariantContextWriter = {
+    val inputHeader = vcfReader.getFileHeader
+
+    val builder = new VariantContextWriterBuilder()
+      .setOutputFile(path.toFile)
+      .setReferenceDictionary(inputHeader.getSequenceDictionary)
+      .setOption(Options.INDEX_ON_THE_FLY)
+      .modifyOption(Options.ALLOW_MISSING_FIELDS_IN_HEADER, allowMissingFieldsInVcfHeader)
+    val writer: VariantContextWriter = builder.build
+
+    val outHeader = if (addPhasedHeaderLines) {
+      // get the header lines to skip
+      val headerLinesToSkip = HapCutCallGenotypeInfo.formatHeaderKeys
+        .filter { key => inputHeader.hasFormatLine(key) }
+        .map { key => inputHeader.getFormatHeaderLine(key) }
+      // get all the header lines from the input
+      val headerLines: util.Set[VCFHeaderLine] = new util.HashSet[VCFHeaderLine](
+        inputHeader.getMetaDataInSortedOrder.filter { line =>
+          if (headerLinesToSkip.contains(line)) {
+            logger.debug(s"Will overwrite header line: '${line.toString}")
+            false
+          }
+          else true
+        }.toSet
+      )
+      // add standard header lines
+      VCFStandardHeaderLines.addStandardFormatLines(headerLines, false, Genotype.PRIMARY_KEYS)
+      // create the header
+      val header: VCFHeader = new VCFHeader(headerLines, inputHeader.getSampleNamesInOrder)
+      // add the new header lines
+      HapCutCallGenotypeInfo.formatHeaderLines.foreach(header.addMetaDataLine)
+      header
+    }
+    else {
+      inputHeader
+    }
+    writer.writeHeader(outHeader)
+    writer
+  }
+}
+
+/** An iterator over all variants in the provided VCFFileReader.  The `input` is the output of HapCut.
```

Comment:
This line is confusing on first (and second) read.  I would suggest killing `The input is the output of HapCut.` from here, and adding it back as a `@param` doc line to the class.